linear-algebra/src/main/scala/scalan/linalgebra/Matrices.scala

```diff
@@ -24,14 +32,18 @@ trait Matrices extends Vectors { self: MatricesDsl =>
     def apply(row: Rep[Int]): Vec[T]
     def apply(row: Rep[Int], column: Rep[Int]): Rep[T]
 
-    //def mapBy[R: Elem](f: Rep[T => R @uncheckedVariance]): Matrix[R]
-    def mapBy[R: Elem](f: Rep[Vector[T] => Vector[R] @uncheckedVariance]): Matr[R]
+    //def mapBy[R: Elem](f: Rep[T => R @uncheckedVariance]): Matr[R]
```

Comment:
Ah, OK. But it'll be a big feature. In case of sparse matrix or, like, `DiagonalMatrix` your example will produce a dense matrix. And sometimes it won't (like if `x => x * 2`). How do we account for it (if we do)?